node/sc/sub_bridge_handler.go

```diff
@@ -344,6 +344,10 @@ func (sbh *SubBridgeHandler) writeServiceChainTxReceipts(bc *blockchain.BlockCha
 	for _, receipt := range receipts {
 		txHash := receipt.TxHash
 		if tx := sbh.subbridge.GetBridgeTxPool().Get(txHash); tx != nil {
+			if receipt.Status != types.ReceiptStatusSuccessful {
+				logger.Error("Transaction's receipt sent from child chain is invalid",
```

Comment:
Okay. Do you have other suggestions or concerns?

Reply:
What do you mean by "removed"?

I understood that the failed txs will be resent and reexecuted until it succeeds, right?

Reply:
No. I was confused with the other stuff. The failed txs are not resent. The only txs in the bridge tx pool (a dedicated tx pool for value transfers) are resent. Otherwise, no tx is recent (e.g., removed tx from the pool). The current implementation unconditionally remove the tx as long the corresponding receipt arrives. We may change to resend failed tx, but those resent tx are not valid due to the low nonce. It leads to stack pending txs. Please refer to the fix at #1295.

FYI, there are two resending logics, but they seems meaningless. The first one was described above. The second one is triggered when making a tx for value transfer handle, not executing the tx. If the failed txs get to be handled again in the counterpart chain, the value transfer recovery is only one correct answer, which finds failed event and makes a new valid tx (i.e., no low nonce issue happened).

Reply:
So, I see that the recovery logic should be revised.
This suggestion is just for the message, what do you think my suggestion? "invalid" is not appropriate based on your description?

Reply:
What do you mean? 
> So, I see that the recovery logic should be revised.

The suggestion is applied.

Reply:
> What do you mean?
> 
> > So, I see that the recovery logic should be revised.

As you described, "there are two resending logics, but they seems meaningless.", so isn't it revised? of course this should be done in another PR.

I saw the the suggestion is applied. I resolve this comment.